File: application-ver2/app/src/main/java/com/example/heartbeat/data/TempData.java
package com.example.heartbeat.data;

/**
 * setPacket()메서드를 호출하면, processData 메서드를 이용해서
 * field 에 대응되는 값들을 처리해서 저장
 */

public class TempData {
    private byte[] packet;
    private int count;
    private double value;

    public void setPacket(byte[] packet) {
        this.packet = packet;
        processData();
    }

    public void setCount(int count) {
        this.count = count;
    }

    public void setValue(float value) {
        this.value = value;
    }

    public byte[] getPacket() {
        return packet;
    }

    public int getCount() {
        return count;
    }

    public double getValue() {
        return value;
    }

    private void processData(){
        this.count = packet[1] & 0xff;
        this.value = (((packet[3] & 0xff) << 8 ) + (packet[2] & 0xff))/100.0;
    }
}

File: Application_part/project_kotlin_version/Kotlin_ver/app/src/main/java/com/example/kotlin_ver/Temperature/TempData.kt
package com.example.kotlin_ver.Temperature

class TempData(packet: ByteArray) {
    var count = 0
    var value = 0.0

    init {
        count = packet[1].toInt() and 0xff
        value = ((packet[3].toInt() and 0xff shl 8) + (packet[2].toInt() and 0xff)) / 100.0
    }

    fun convertCelciustoF(): Double {
        return value * (9 / 5.0) + 32.0
    }

}